main/src/com/miloshpetrov/sol2/game/chunk/ChunkFiller.java

```diff
@@ -163,83 +172,162 @@ private void fillAsteroids(SolGame game, RemoveController remover, boolean forBe
       float rotSpd = SolMath.rnd(JUNK_MAX_ROT_SPD);
+
+      // Create the resulting sprite and add it to the list as the only element
       RectSprite s = new RectSprite(tex, sz, 0, 0, new Vector2(), DraLevel.DECO, SolMath.rnd(180), rotSpd, SolColor.LG, false);
       ArrayList<Dra> dras = new ArrayList<Dra>();
       dras.add(s);
 
+      // Create a FarDras instance for this piece of junk and only allow it to be drawn when it's not hidden by a planet
       Vector2 spd = new Vector2();
       SolMath.fromAl(spd, SolMath.rnd(180), SolMath.rnd(JUNK_MAX_SPD_LEN));
       FarDras so = new FarDras(dras, junkPos, spd, remover, true);
+      // Add the object to the object manager
       game.getObjMan().addFarObjNow(so);
     }
   }
 
+  /**
+   * Add specks of dust to the background layer closest to the front.
+   * <p/>
+   * Dust is fixed in the world and therefore moves opposite to the cameras movement.
+   *
+   * @param game     The {@link SolGame} instance to work with
+   * @param chCenter The center of the chunk
+   * @param remover
+   */
   private void fillDust(SolGame game, Vector2 chCenter, RemoveController remover) {
     ArrayList<Dra> dras = new ArrayList<Dra>();
     int count = getEntityCount(DUST_DENSITY);
     if (count == 0) return;
+
     TextureAtlas.AtlasRegion tex = myDustTex;
     for (int i = 0; i < count; i++) {
+      // Select a random position in the chunk centered around chCenter, relative to the position of the chunk.
       Vector2 dustPos = getRndPos(chCenter);
       dustPos.sub(chCenter);
+      // Create the resulting sprite and add it to the list
       RectSprite s = new RectSprite(tex, DUST_SZ, 0, 0, dustPos, DraLevel.DECO, 0, 0, SolColor.W, false);
       dras.add(s);
     }
+
+    // Create a common FarDras instance for the specks of dust and only allow the dust to be drawn when it's not hidden by a planet
     FarDras so = new FarDras(dras, chCenter, new Vector2(), remover, true);
     game.getObjMan().addFarObjNow(so);
   }
 
+  /**
+   * Find a random position in a chunk centered around chCenter, relative to the entire map, and make sure it is not yet
+   * occupied by another entity.
+   * <p/>
+   * Up to 100 tries will be made to find an unoccupied position; if by then none has been found, <code>null</code> will be returned.
+   *
+   * @param g        The {@link SolGame} instance to work with
+   * @param chCenter The center of a chunk in which a random position should be found
+   * @return A random, unoccupied position in a chunk centered around chCenter, relative to the entire map, or <code>null</code> if within 100 tries no unoccupied position has been found
+   */
   private Vector2 getFreeRndPos(SolGame g, Vector2 chCenter) {
-    Vector2 pos = new Vector2(chCenter);
     for (int i = 0; i < 100; i++) {
-      pos.x += SolMath.rnd(Const.CHUNK_SIZE/2);
-      pos.y += SolMath.rnd(Const.CHUNK_SIZE/2);
+      Vector2 pos = getRndPos(new Vector2(chCenter));

```

Comment:
Does this change work as expected? Previously it was adding a random amount to x and y positions every iteration in the for loop. After the change you are getting a new vector position every iteration.